controller/internal/enforcer/nfqdatapath/nfq_windows.go

```diff
@@ -0,0 +1,165 @@
+// +build windows
+
+package nfqdatapath
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"go.aporeto.io/trireme-lib/controller/internal/enforcer/nfqdatapath/afinetrawsocket"
+	"go.aporeto.io/trireme-lib/controller/internal/enforcer/nfqdatapath/nflog"
+	"go.aporeto.io/trireme-lib/controller/internal/windows/frontman"
+	"go.aporeto.io/trireme-lib/controller/pkg/connection"
+	"go.aporeto.io/trireme-lib/controller/pkg/packet"
+	"go.uber.org/zap"
+)
+
+func (d *Datapath) startFrontmanPacketFilter(ctx context.Context, nflogger nflog.NFLogger) error {
+	driverHandle, err := frontman.GetDriverHandle()
+	if err != nil {
+		return err
+	}
+
+	nflogWin := nflogger.(*nflog.NfLogWindows)
+
+	packetCallback := func(packetInfoPtr, dataPtr uintptr) uintptr {
+
+		packetInfo := *(*frontman.PacketInfo)(unsafe.Pointer(packetInfoPtr))
+		packetBytes := (*[1 << 30]byte)(unsafe.Pointer(dataPtr))[:packetInfo.PacketSize:packetInfo.PacketSize]
+
+		var packetType int
+		if packetInfo.Outbound != 0 {
+			packetType = packet.PacketTypeApplication
+		} else {
+			packetType = packet.PacketTypeNetwork
+		}
+
+		// Parse the packet
+		mark := int(packetInfo.Mark)
+		parsedPacket, err := packet.New(uint64(packetType), packetBytes, strconv.Itoa(mark), true)
+
+		parsedPacket.WindowsMetadata = &afinetrawsocket.WindowsPacketMetadata{PacketInfo: packetInfo, IgnoreFlow: false}
+		var processError error
+		var tcpConn *connection.TCPConnection
+		var udpConn *connection.UDPConnection
+		if err != nil {
+			parsedPacket.Print(packet.PacketFailureCreate, d.packetLogs)
+		} else if parsedPacket.IPProto() == packet.IPProtocolTCP {
+			if packetType == packet.PacketTypeNetwork {
+				tcpConn, processError = d.processNetworkTCPPackets(parsedPacket)
+			} else {
+				tcpConn, processError = d.processApplicationTCPPackets(parsedPacket)
+			}
+		} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+			// process udp packet
+			if packetType == packet.PacketTypeNetwork {
+				udpConn, processError = d.ProcessNetworkUDPPacket(parsedPacket)
+			} else {
+				udpConn, processError = d.ProcessApplicationUDPPacket(parsedPacket)
+			}
+		} else {
+			processError = fmt.Errorf("invalid ip protocol: %d", parsedPacket.IPProto())
+		}
+
```

Comment:
This logic seems a little convoluted, and I am not sure how it will work. 

If err!=nil, we print it, but not drop it or return. processError remains nil, and thus the subsequent if statement is not met. 

I would suggest we explicitly define on error and then return or drop (or whatever we need to do ), and then simplify the nested IFs.

Reply:
I believe this logic is similar to that of linux/master. Can you verify?

Reply:
https://github.com/aporeto-inc/trireme-lib/blob/master/controller/internal/enforcer/nfqdatapath/nfq_linux.go#L77

Reply:
will simplify this code (linux and windows) in another branch and/or after this merge